ctapipe/utils/rgbtohex_c.cc

```diff
@@ -0,0 +1,18 @@
+/*
+C extension to convert and rgb array into a array of chars containing the
+hexidecimal string equivalent.
+*/
+
+#include <iostream>
+#include <stdint.h>
+#include <stdio.h>
+
+extern "C" void rgbtohex(const uint8_t* rgb, size_t n_pix, char* hex)
```

Comment:
TL;DR: I believe in this particular case the C-extension is **not necessary**.

I believe we should avoid C-extensions as long as possible and I assume you believe so too (as opposed to writing C-extension just for fun). So I assume you had no chance to avoid this, since with pure Python this was performing horrible. In such a case I believe C-extensions are fine.

Still, I personally consider writing a C-extension so exceptional that a little discussion of it might not be too annoying. I would have liked the author to present maybe some numbers in a comment or in the PR-discussion to backup the decision to go for a C-extension. 

Since these numbers were missing, I tried to understand myself what was so slow here. For this I created a Python implementation of what I understood the C-implementation to do and profiled it by hand (using simply `%timeit` in my ipython session).

The result was a second implementation of `intensity_to_hex` which is simply named `intensity_to_hex2` which is pure python and happens to be even faster than the version using the C-extension. 

----

Here is the comparison:
```
In [1]: from rgbtohex import *

In [2]: array = np.random.rand(10000)

In [3]: assert (intensity_to_hex(array) == intensity_to_hex2(array)).all()

In [4]: %timeit intensity_to_hex(array)
2.99 ms ± 13.7 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [5]: %timeit intensity_to_hex2(array)
1.75 ms ± 10.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

This is the original `intensity_to_hex` (rewritten a bit and added the runtimes per line):
```python
def intensity_to_hex(array, minval=None, maxval=None):
    hex_ = np.empty((array.size, 8), dtype='S1')   #   <1 us
    rgb = intensity_to_rgb(array, minval, maxval)  #  200 us
    rgbtohex(rgb, array.size, hex_)                # 1300 us
    return hex_.view('S8').astype('U8')[:, 0]      # 1300 us
```

This is the other variant:
```python
def intensity_to_hex2(array, minval=None, maxval=None):
    import codecs
    hex_ = np.zeros((array.size, 8), dtype='B')   #    3 us
    rgb = intensity_to_rgb(array, minval, maxval) #  200 us

    hex_encoded = codecs.encode(rgb, 'hex')       #   39 us
    bytes_ = np.frombuffer(hex_encoded, 'B')      #   <1 us
    bytes_2d = bytes_.reshape(-1, 8)              #   <1 us
    hex_[:, 0] = ord('#')                         #    4 us
    hex_[:, 1:7] = bytes_2d[:, 0:6]               #   48 us

    return hex_.view('S8').astype('U8')[:, 0]     # 1300 us
```

Reply:
This is great! 

Yeah, I couldn't understand how to do this efficiently in Python (I had to use a for loop), so I resorted to the C extension. I love your solution. Thanks!